src/components/UserMenu/UserMenu.tsx

```diff
@@ -1,227 +1,139 @@
+
+  useEffect(() => {
+    setIsOpen(isOpen)
     setTimeout(() => {
-      if (this.mounted) {
-        this.setState({ isClickable: value })
+      if (isMounted) {
+        setIsClickable(isOpen)
       }
     }, 250)
-  }
-
-  componentDidMount(): void {
-    this.mounted = true
-  }
-
-  componentWillUnmount(): void {
-    this.mounted = false
-  }
-
-  renderManaBalances = (): React.ReactNode => {
-    const { manaBalances, onClickBalance } = this.props
-
-    return (
-      <span className="dcl account-wrapper">
-        {Object.keys(manaBalances).map((network) => (
-          <Mana
-            key={network}
-            network={network as Network}
-            size="small"
-            className={onClickBalance ? 'clickable' : undefined}
-            title={`${manaBalances[network].toLocaleString()} MANA`}
-            href={config.get('ACCOUNT_URL')}
+  }, [isOpen, setIsOpen, setIsClickable, setIsMounted])
+
+  useEffect(() => {
+    setIsMounted(true)
+    return () => setIsMounted(false)
+  }, [])
+
+  const handleToggle = useCallback(
+    (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+      const trackId = uuidv4()
+      setIsOpen((prev) => {
+        !prev && setTrackingId(trackId)
+        !prev && onClickOpen && onClickOpen(event, trackId)
+        return !prev
+      })
+    },
+    [setIsOpen, onClickOpen]
+  )
+
+  const handleClose = useCallback(() => {
+    setIsOpen(false)
+  }, [setIsOpen])
+
+  const handleClickJumpIn = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+      onClickMenuItem &&
+        onClickMenuItem(event, UserMenuLabels.PROFILE, trackingId)
+
+      setTimeout(
+        () => {
+          onClickJumpIn
+            ? onClickJumpIn(event)
+            : window.open(config.get('EXPLORER_URL'), '_blank', 'noopener')
+        },
+        onClickMenuItem ? 300 : 0
+      )
+    },
+    [onClickJumpIn, onClickMenuItem, trackingId]
+  )
+
+  const isSomeBalanceTooHigh = useMemo(
+    () =>
+      manaBalances &&
+      Object.values(manaBalances).some(
+        (balance) => Number(balance.toFixed(2)).toLocaleString().length > 5
+      ),
+    [manaBalances]
+  )
+
+  return (
+    <Column align="right">
+      <Row className={classNames('dcl', 'user-menu-wrapper')}>
+        <div
+          className={classNames('dcl', 'user-menu')}
+          onBlur={handleClose}
+          tabIndex={0}
+        >
+          {isSignedIn && (
+            <UserMenuSignedIn
+              {...signInProps}
+              manaBalances={manaBalances}
+              trackingId={trackingId}
+              isOpen={isOpen}
+              isClickable={isClickable}
+              onClickToggle={handleToggle}
+              onClickMenuItem={onClickMenuItem}
+              onClickBalance={onClickBalance}
+            />
+          )}
+          {!isSignedIn && (
+            <Button inverted disabled={isSigningIn} onClick={onClickSignIn}>
+              {UserMenuLabels.SIGN_IN}
```

Comment:
When it comes to translations, we usually name the props i18n, would you mind naming it as `i18n` as well?

Reply:
I'll change all this into a new file i18n